modules/library/main/src/main/java/org/geotools/filter/spatial/ReprojectingFilterVisitor.java

```diff
@@ -103,7 +111,11 @@ public Object visit(BBOX filter, Object extraData) {
         } else if (filter.getExpression2() instanceof PropertyName) {
             propertyName = (PropertyName) filter.getExpression2();
         }
-        CoordinateReferenceSystem targetCrs = findPropertyCRS(propertyName);
+        CoordinateReferenceSystem targetCrs = this.targetCrs;
```

Comment:
Why make it BBOX specific, would it make sense to move this inside findPropertyCRS instead, so that geometry literals and other comparisons get reprojected as well?

Reply:
Good point.